### PR TITLE
Feature template deserialize

### PR DESCRIPTION
**Pull request by PrzeG**
# Pull Request summary:
Adjustments to Feature Template deserialization, adding proper support for nested values and correct field names

# Description of changes:

# Checklist:
- [ ] Make sure to run pre-commit before committing changes
- [ ] Make sure all checks have passed
- [ ] PR description is clear and comprehensive
- [ ] Mentioned the issue that this PR solves (if applicable)
- [ ] Make sure you test the changes

### CODE REVIEW

**Comment by jpkrajewski** [comment]
Please add unit tests

File: catalystwan/utils/feature_template/find_template_values.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+from typing import Dict, List, Optional, Union
+
+from catalystwan.api.templates.device_variable import DeviceVariable
+
+
+def find_template_values(
```
Comment (jpkrajewski) [inline catalystwan/utils/feature_template/find_template_values.py:6]: The find_template_values function is integral to the UX2 transformation process. It's imperative to conduct thorough unit tests to ensure its reliability and accuracy. Testing should encompass both default Cisco models and fully populated models.

File: catalystwan/integration_tests/test_find_template_values.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            username=cast(str, os.environ.get("TEST_VMANAGE_USERNAME")),
+            password=cast(str, os.environ.get("TEST_VMANAGE_PASSWORD")),
+        )
+        self.templates = self.session.api.templates._get_feature_templates()
+
+    def test_find_template_value(self):
+        for template in self.templates:
+            definition = template.template_definiton
+            with self.subTest(template_name=template.name):
+                parsed_values = find_template_values(definition)
```
Comment (jkrajew) [inline catalystwan/integration_tests/test_find_template_values.py:23]: Address the underlying issues causing the test function to fail for our demo vManage.

**Review by jkrajew — CHANGES_REQUESTED** [review]

File: catalystwan/integration_tests/test_find_template_values.py (hunk 2 above)
Comment (jkrajew) [reply catalystwan/integration_tests/test_find_template_values.py:23]: Please consider to write this as unit test to not rely on having fully configured vManage

**Review by sbasan — APPROVED** [review]

**Review by jkrajew — APPROVED** [review]
It's working for Demo vManage

image

**Pull request merged**